relayer/socket/server.go

```diff
@@ -89,292 +88,256 @@ func (s *Server) send(conn net.Conn, data []byte) error {
 	case EventGetMessageList:
 		req := new(ReqMessageList)
-		if err := jsoniter.Unmarshal(msg.Data, req); err != nil {
-			return nil, err
-		}
-		store := s.rly.GetMessageStore()
-		messages, err := store.GetMessages(req.Chain, req.Pagination)
-		if err != nil {
-			return nil, err
+		if err := jsoniter.Unmarshal(data, req); err != nil {
+			return response.SetError(err)
 		}
-		total, err := store.TotalCountByChain(req.Chain)
+		msgStore := s.rly.GetMessageStore()
+		messages, err := msgStore.GetMessages(req.Chain, &store.Pagination{Limit: req.Limit})
 		if err != nil {
-			return nil, err
+			return response.SetError(err)
 		}
-		data, err := jsoniter.Marshal(&ResMessageList{messages, int(total)})
+		total, err := msgStore.TotalCountByChain(req.Chain)
 		if err != nil {
-			return nil, err
+			return response.SetError(err)
 		}
-		return &Message{EventGetMessageList, data}, nil
+		return response.SetData(&ResMessageList{messages, int(total)})
 	case EventMessageRemove:
 		req := new(ReqMessageRemove)
-		if err := jsoniter.Unmarshal(msg.Data, req); err != nil {
-			return nil, err
+		if err := jsoniter.Unmarshal(data, req); err != nil {
+			return response.SetError(err)
 		}
 		store := s.rly.GetMessageStore()
 		key := &types.MessageKey{Src: req.Chain, Sn: req.Sn}
 		message, err := store.GetMessage(key)
 		if err != nil {
-			return nil, err
+			return response.SetError(err)
 		}
 		if err := store.DeleteMessage(key); err != nil {
-			return nil, err
+			return response.SetError(err)
 		}
-		data, err := jsoniter.Marshal(&ResMessageRemove{req.Sn, req.Chain, message.Dst, message.MessageHeight, message.EventType})
-		if err != nil {
-			return nil, err
-		}
-		return &Message{EventMessageRemove, data}, nil
+		return response.SetData(&ResMessageRemove{req.Sn, req.Chain, message.Dst, message.MessageHeight, message.EventType})
 	case EventRelayMessage:
 		req := new(ReqRelayMessage)
-		if err := jsoniter.Unmarshal(msg.Data, req); err != nil {
-			return nil, err
+		if err := jsoniter.Unmarshal(data, req); err != nil {
+			return response.SetError(err)
 		}
-
 		src, err := s.rly.FindChainRuntime(req.Chain)
 		if err != nil {
-			return nil, err
+			return response.SetError(err)
 		}
-
-		if req.Height != 0 {
-			msgs, err := src.Provider.GenerateMessages(context.Background(), types.NewMessagekeyWithMessageHeight(&types.MessageKey{Src: req.Chain, Sn: req.Sn}, req.Height))
-			if err != nil {
-				return nil, err
-			}
-			for _, msg := range msgs {
-				src.MessageCache.Add(types.NewRouteMessage(msg))
-			}
-			data, err := jsoniter.Marshal(&ResRelayMessage{types.NewRouteMessage(msgs[0])})
-			if err != nil {
-				return nil, err
-			}
-			return &Message{EventRelayMessage, data}, nil
+		if req.FromHeight > req.ToHeight {
+			return response.SetError(fmt.Errorf("fromHeight should be less than toHeight"))
+		} else if req.ToHeight-req.FromHeight > 100 {
+			return response.SetError(fmt.Errorf("fromHeight and toHeight should be less than 100"))
```

Comment:
the error message is misleading please correct.

Reply:
fixed